src/Injection/LazyArray.php

```diff
@@ -0,0 +1,62 @@
+<?php
+/**
+ *
+ * This file is part of Aura for PHP.
+ *
+ * @license http://opensource.org/licenses/MIT MIT
+ *
+ */
+namespace Aura\Di\Injection;
+
+/**
+ *
+ * Returns the value of a callable when invoked (thereby invoking the callable).
+ *
+ * @package Aura.Di
+ *
+ */
+class LazyArray implements LazyInterface
+{
+    /**
+     *
+     * Array of callables to invoke.
+     *
+     * @var array
+     *
+     */
+    protected $callables;
+
+    /**
+     *
+     * Constructor.
+     *
+     * @param array $callables The callables to invoke.
+     *
+     */
+    public function __construct(array $callables)
+    {
+        $this->callables = $callables;
+    }
+
+    /**
+     *
+     * Invokes the array of closures to create the instance array.
+     *
+     * @return array The array of objects created by the closures.
+     *
+     */
+    public function __invoke()
+    {
+        // convert Lazy objects in the callables
+        if (is_array($this->callables)) {
```

Comment:
I wonder what will be the real use case for people to extend the class. In that case I believe the `__invoke` would also be altered by them ;-) if they modify the constructor for it may be accepting different values also.

I would suggest to remove the check, but lets do it only after hearing from @pmjones .

Thank you